### ISSUE TITLE
[IMPROVEMENT] Importing Cities from JSON file

### ISSUE THREAD

**Issue by nscharrenberg** [nscharrenberg/COVID-21#37]
**What could be improved?**
Efficiency and general flow of importing cities from a JSON file.

**What is currently wrong with that which should be improved?**
Seems to use to many loops, adding up to time complexity.

**How could we achieve this?**
Assuming you have:
```json
{
      "name": "San Francisco",
      "color": "Blue",
      "position": [-650,200,1],
      "population": 874961,
      "neighbours": [
        "Chicago",
        "Mexico City",
        "Calgary",
        "Tokyo"
      ]
    },
 ```
 
 I would see the steps someting like this:
 
 1. Instantiate a temporary neighbours arraylist called `tempNeighbors`
 2. Instantiate a cities array called `cities`
 2. Read the file as `fileCities`, which is an array
 3. loop through `fileCities` array
 3.1 Instantiate a new city and add it to the `cities` array
 3.2 add edges to `tempNeighbors` array/hashmap.
 4. loop through `tempNeighbors` array/hashmap
 4.1 `cities.get(key).addNeighbor(cities.get(neighorKey))`
 5. return cities.

So basically 2 (maybe 3) for loops instead of the 4 for loops we have now.
